miden-lib/asm/miden/kernels/tx/account.masm

```diff
@@ -469,39 +472,57 @@ export.set_map_item.3
     # => [OLD_MAP_ROOT, OLD_VALUE, ...]
 end
 
-#! Verifies that the procedure root is part of the account code Merkle tree. Panics if the
-#! procedure root is not part of the account code Merkle tree.
+#! Returns the procedure information
+#!
+#! Stack: [index, ...]
+#! Output: [PROC_ROOT, storage_offset, ...]
+#!
+#! - PROC_ROOT is the hash of the procedure.
+#! - storage_offset is the procedure storage offset.
+#!
+#! Panics if
+#! - index is out of bounds
+export.get_procedure_info
+  # check that index < number of procedures contained in the account code
+  dup exec.memory::get_num_account_procedures gt assert
+  # => [index]
+
+  # get procedure section ptr
+  push.2 mul exec.memory::get_account_procedures_section_offset add dup push.1 add swap
+  # => [proc_ptr, offset_ptr]
+
+  # load procedure information from memory
+  padw movup.4 mem_loadw movup.4 mem_load movdn.4
+  # => [PROC_ELEMENTS, storage_offset]
```

Comment:
A slightly more efficient way to do this would be:
```
# get procedure pointers
push.2 mul exec.memory::get_account_procedures_section_offset add dup push.1 add
# => [storage_offset_ptr, proc_root_ptr]

# load procedure information from memory
mem_load swap padw movup.4 mem_loadw
# => [PROC_ELEMENTS, storage_offset]
```

Reply:
Replaced.